compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -642,6 +673,24 @@ impl DataFlowGraph {
             false
         }
     }
+
+    pub(crate) fn is_global(&self, value: ValueId) -> bool {
+        matches!(self.values[value], Value::Global(_))
+    }
+
+    pub(crate) fn get_instruction(&self, value: ValueId) -> Option<&Instruction> {
+        match &self[value] {
```

Comment:
I think this method needs some clarification in both a doc comment and making the name more verbose to highlight that the returned instruction may not be in this DFG (!) since using the valueids directly from it could result in different underlying values from this DFG. As-is I can see someone using this as a general purpose way to get an instruction in an optimization pass and run into that bug without knowing - and it'd be hard to track down.

Reply:
I'll add a safety comment either way. 

How does the name `get_local_or_global_instruction` sound?

Reply:
I made the rename to `get_local_or_global_instruction` with an assertion for `MakeArray` when we have a global instruction. I also included a small comment stating that this may come from a separate DFG.